cmd/configschema/go.mod

```diff
@@ -2,7 +2,7 @@
 // See https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/30187
 module github.com/open-telemetry/opentelemetry-collector-contrib/cmd/configschema
 
-go 1.21
+go 1.21.9
```

Comment:
This should be reverted, same for other `go.mod`

Reply:
Got it, it's probably fine to use 1.21.9 now with https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32160 @mx-psi WDYT?

Reply:
I don't think so, I just tried it with Go 1.21
```
❯ go mod tidy        
go: go.mod requires go >= 1.21.9 (running go 1.21.0)
```

Reply:
@mx-psi and I discussed offline - can we modify the Agent go.mod to 1.21 for now? We think it's safer to do so

Reply:
Just finished changing both repos to go 1.21.0